drivers/led/is31fl3194/is31fl3194.c

```diff
@@ -0,0 +1,235 @@
+	const struct led_info *leds_info;
+
+	leds_info = is31fl3194_led_to_info(config, led);
+	if (!leds_info) {
+		*info = NULL;
+		return -EINVAL;
+	}
+
+	*info = leds_info;
+
+	return 0;
+}
+
+int is31fl3194_set_brightness(const struct device *dev, uint32_t led, uint8_t value)
+{
+	const struct is31fl3194_config *config = dev->config;
+	struct is31fl3194_ctx *ctx = (struct is31fl3194_ctx *)&config->ctx;
+	struct is31fl3194_outx_cl outx_cl = { .cl = value };
+	int res;
+
+	res = is31fl3194_outx_cl_set(ctx, led, outx_cl);
+	if (res) {
+		return res;
+	}
+
+	return is31fl3194_color_update(ctx);
+}
+
+int is31fl3194_write_channels(const struct device *dev, uint32_t start_channel,
+			      uint32_t num_channels, const uint8_t *buf)
+{
+	int res;
+
+	if ((start_channel + num_channels) > IS31FL3194_INDEX_MAX) {
+		return -EINVAL;
+	}
+
+	for (int i = 0; i < num_channels; i++) {
+		res = is31fl3194_set_brightness(dev, start_channel + i, buf[i]);
+		if (res) {
+			return res;
+		}
+	}
+
+	return 0;
+}
+
+static const struct led_driver_api is31fl3194_api = {
+	/* Mandatory callbacks. */
+	.on = is31fl3194_on,
+	.off = is31fl3194_off,
+	/* Optional callbacks. */
+	.get_info = is31fl3194_get_info,
+	.set_brightness = is31fl3194_set_brightness,
+	.write_channels = is31fl3194_write_channels,
+};
+
+static int is31fl3194_led_init(const struct device *dev)
+{
+	const struct is31fl3194_config *config = dev->config;
+	struct is31fl3194_ctx *ctx = (struct is31fl3194_ctx *)&config->ctx;
+	int res;
+	uint8_t val;
+
+	if (!device_is_ready(config->i2c.bus)) {
+		LOG_ERR("I2C bus is not ready");
+		return -ENODEV;
+	}
+
+	res = is31fl3194_product_id_get(ctx, &val);
+	if (res) {
+		LOG_ERR("failed to read Product ID: %d", res);
+		return res;
+	}
+
+	res = is31fl3194_reset(ctx);
+	if (res) {
+		LOG_ERR("failed to reset device: %d", res);
+		return res;
+	}
+
+	for (int i = 0; i < config->num_leds; i++) {
+		enum is31fl3194_current_band current_band;
+
+		if (config->curr_map_max[i] == 10) {
+			current_band = IS31FL3194_CURRENT_BAND_1;
+		} else if (config->curr_map_max[i] == 20) {
+			current_band = IS31FL3194_CURRENT_BAND_2;
+		} else if (config->curr_map_max[i] == 30) {
+			current_band = IS31FL3194_CURRENT_BAND_3;
+		} else {
+			current_band = IS31FL3194_CURRENT_BAND_4;
+		}
+
+		res = is31fl3194_current_band_set(ctx, config->leds_info[i].index, current_band);
+		if (res) {
+			LOG_ERR("failed to set current band: %d", res);
+			return res;
+		}
+	}
+
+	return is31fl3194_ops_ssd_set(ctx, 1);
+}
+
+#define COLOR_MAPPING(led_node_id)	\
+	const uint8_t color_mapping_##led_node_id[] = DT_PROP(led_node_id, color_mapping);
+
+#define LED_INFO(led_node_id)	\
+	{	\
+		.label = DT_LABEL(led_node_id),	\
+		.index = DT_PROP(led_node_id, index),	\
+		.num_colors = DT_PROP_LEN(led_node_id, color_mapping),	\
+		.color_mapping = color_mapping_##led_node_id,	\
+	},
+
+#define IS31FL3194_CURR_BAND_MAX(id) DT_PROP_OR(id, curr_map_max, 20),
```

Comment:
I don't think it is working. The property is named `curr-band-max` in the DT binding. So it should be `curr_band_max` here.